src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -14644,7 +14644,30 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
                     ASR::Array_t* arr = ASR::down_cast<ASR::Array_t>(
                         ASRUtils::type_get_past_allocatable_pointer(
                             ASRUtils::expr_type(x.m_args[i])));
-                    if(arr->m_physical_type != ASR::array_physical_typeType::PointerArray){
+                    //Copy only when we have DescriptorArray Type, where not referencing through pointer
+                    //TODO: Check for non-unit strides, and create a copy only when having non-unit strides.
+                    bool copy_needed = (arr->m_physical_type == ASR::array_physical_typeType::DescriptorArray) &&
+                                        ASR::is_a<ASR::Var_t>(*x.m_args[i]);
+                    if(copy_needed){
+                        // For DescriptorArray, we need to create a contiguous copy
+                        // because the descriptor may have non-unit strides (e.g., from array sections)
+                        // and the backend expects properly strided elements in output
```

Comment:
Is this what you meant to say?
```suggestion
                        // and the string format runtime function expects contiguous arrays for inputs
```
Or maybe I misunderstand the issue.